EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityInfoDAO.java

```diff
@@ -1,16 +1,415 @@
+import eu.neclab.ngsildbroker.commons.datatypes.RegistrationEntry;
+import eu.neclab.ngsildbroker.commons.datatypes.RemoteHost;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.AppendEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.CreateEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.DeleteAttributeRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.DeleteEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.UpdateEntityRequest;
+import eu.neclab.ngsildbroker.commons.enums.ErrorType;
+import eu.neclab.ngsildbroker.commons.exceptions.ResponseException;
+import eu.neclab.ngsildbroker.commons.storage.ClientManager;
+import eu.neclab.ngsildbroker.commons.tools.HttpUtils;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.tuples.Tuple2;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.MultiMap;
+import io.vertx.mutiny.sqlclient.Row;
+import io.vertx.mutiny.sqlclient.RowIterator;
+import io.vertx.mutiny.sqlclient.RowSet;
+import io.vertx.mutiny.sqlclient.Tuple;
+import io.vertx.pgclient.PgException;
 
 @Singleton
-public class EntityInfoDAO extends StorageDAO {
+public class EntityInfoDAO {
+
+	private static Logger logger = LoggerFactory.getLogger(EntityInfoDAO.class);
+
+	@Inject
+	ClientManager clientManager;
+
+	GeoJSONReader geoReader = new GeoJSONReader(JtsSpatialContext.GEO, new SpatialContextFactory());
 
-	@Override
-	protected StorageFunctionsInterface getStorageFunctions() {
-		return new EntityStorageFunctions();
+	public Uni<Void> batchCreateEntity(List<CreateEntityRequest> requests) {
+		return clientManager.getClient(requests.get(0).getTenant(), true).onItem().transformToUni(client -> {
+			List<Tuple> tuples = new ArrayList<>(requests.size());
+			for (CreateEntityRequest request : requests) {
+				tuples.add(Tuple.of(request.getId(),
+						((List<String>) request.getPayload().get(NGSIConstants.JSON_LD_TYPE)).toArray(new String[0]),
+						new JsonObject(request.getPayload())));
+			}
+			return client.preparedQuery(
+					"INSERT INTO ENTITY(ID,E_TYPES, ENTITY) VALUES ($1, $2, $3) ON CONFLICT DO NOTHING RETURNING ID as notadded")
+					.executeBatch(tuples).onFailure().recoverWithUni(e -> {
+						if (e instanceof PgException) {
+							if (((PgException) e).getCode().equals(AppConstants.SQL_ALREADY_EXISTS)) {
+								return Uni.createFrom().failure(new ResponseException(ErrorType.AlreadyExists));
+							}
+						}
+						return Uni.createFrom().failure(e);
+					}).onItem().transformToUni(v -> {
+						System.out.println(v);
+						return Uni.createFrom().voidItem();
+					});
+		});
 	}
+
+	public Uni<Map<String, Object>> partialUpdateAttribute(UpdateEntityRequest request) {
+		return clientManager.getClient(request.getTenant(), false).onItem().transformToUni(client -> {
+			List<Map<String, Object>> effectivePayloads = (List<Map<String, Object>>) request.getPayload()
```

Comment:
Handle proper exception in case of effectivePayloads .

Reply:
Done